internal/node/machine_mgr.go

```diff
@@ -338,53 +319,17 @@ func (m *MachineManager) LookupMachine(vmId string) *runningFirecracker {
 	return vm
 }
 
-func (m *MachineManager) awaitHandshake(vmid string) {
-	timeoutAt := time.Now().UTC().Add(m.handshakeTimeout)
-
-	handshakeOk := false
-	for !handshakeOk && !m.stopping() {
-		if time.Now().UTC().After(timeoutAt) {
-			m.log.Error("Did not receive NATS handshake from agent within timeout.", slog.String("vmid", vmid))
-			if len(m.handshakes) == 0 {
-				m.log.Error("First handshake failed, shutting down to avoid inconsistent behavior")
-				m.cancel()
-			}
-			return
-		}
-
-		_, handshakeOk = m.handshakes[vmid]
-		time.Sleep(time.Millisecond * agentapi.DefaultRunloopSleepTimeoutMillis)
+func (m *MachineManager) agentHandshakeTimedOut(agentId string) {
```

Comment:
Does this get called anywhere else other than `awaitHandshake`? If not, it makes it harder to read as a separate function.

What will the use-cases be to allow `AgentClient` instances to support this provided timeout callback?